django-backend/fecfiler/web_services/summary/tasks.py

```diff
@@ -19,175 +21,213 @@ def __str__(self):
-    # line 30ai
-    report.form_3x.L30ai_shared_federal_activity_h6_fed_share_period = a.get("line_30ai", 0)  # noqa: E501
-    report.form_3x.L30ai_shared_federal_activity_h6_fed_share_ytd = b.get("line_30ai", 0)
-    # line 30aii
-    report.form_3x.L30aii_shared_federal_activity_h6_nonfed_period = a.get("line_30aii", 0)  # noqa: E501
-    report.form_3x.L30aii_shared_federal_activity_h6_nonfed_ytd = b.get("line_30aii", 0)
-    # line 30b
-    report.form_3x.L30b_nonallocable_fed_election_activity_period = a.get("line_30b", 0)
-    report.form_3x.L30b_nonallocable_fed_election_activity_ytd = b.get("line_30b", 0)
-    # line 30c
-    report.form_3x.L30c_total_federal_election_activity_period = a.get("line_30c", 0)
-    report.form_3x.L30c_total_federal_election_activity_ytd = b.get("line_30c", 0)
-    # line 31
-    report.form_3x.L31_total_disbursements_period = a.get("line_31", 0)
-    report.form_3x.L31_total_disbursements_ytd = b.get("line_31", 0)
-    # line 32
-    report.form_3x.L32_total_federal_disbursements_period = a.get("line_32", 0)
-    report.form_3x.L32_total_federal_disbursements_ytd = b.get("line_32", 0)
-    # line 33
-    report.form_3x.L33_total_contributions_period = a.get("line_33", 0)
-    report.form_3x.L33_total_contributions_ytd = b.get("line_33", 0)
-    # line 34
-    report.form_3x.L34_total_contribution_refunds_period = a.get("line_34", 0)
-    report.form_3x.L34_total_contribution_refunds_ytd = b.get("line_34", 0)
-    # line 35
-    report.form_3x.L35_net_contributions_period = a.get("line_35", 0)
-    report.form_3x.L35_net_contributions_ytd = b.get("line_35", 0)
-    # line 36
-    report.form_3x.L36_total_federal_operating_expenditures_period = a.get("line_36", 0)
-    report.form_3x.L36_total_federal_operating_expenditures_ytd = b.get("line_36", 0)
-    # line 37
-    report.form_3x.L37_offsets_to_operating_expenditures_period = a.get("line_37", 0)
-    report.form_3x.L37_offsets_to_operating_expenditures_ytd = b.get("line_37", 0)
-    # line 38
-    report.form_3x.L38_net_operating_expenditures_period = a.get("line_38", 0)
-    report.form_3x.L38_net_operating_expenditures_ytd = b.get("line_38", 0)
-
-    report.form_3x.save()
-    report.calculation_status = CalculationState.SUCCEEDED
-    report.save()
-    return report.id
+
+    reports_to_recalculate = get_reports_to_calculate(primary_report)
+    calculation_token = uuid.uuid4()
+    reports_to_recalculate.update(
+        calculation_token=calculation_token,
+        calculation_status=CalculationState.CALCULATING
+    )
+    claimed_reports = list(reports_to_recalculate)
+
+    for report in claimed_reports:
+        summary_service = SummaryService(report)
+        summary = summary_service.calculate_summary()
+        a = summary["a"]
+        b = summary["b"]
+
+        # line 6a
+        report.form_3x.L6a_cash_on_hand_jan_1_ytd = b["line_6a"]
+        # line 6b
+        report.form_3x.L6b_cash_on_hand_beginning_period = a["line_6b"]
```

Comment:
Is there a reason I'm missing that the getter ( i.e. a.get("line_6b", 0) ) is not being used on lines 59 and 61?
Do we want a None result in some scenarios for COH?

Reply:
It's my understanding that we do sometimes want to have a None result for COH.  If I remember correctly, *in the front end* we ask the user to provide Cash on Hand if all reports within a year have a COH of None.